lib/elixir/lib/inspect.ex

```diff
@@ -372,6 +390,38 @@ defimpl Inspect, for: Reference do
 end
 
 defimpl Inspect, for: Any do
+  defmacro __deriving__(module, struct, options) do
+    fields =
+      struct
+      |> Map.drop([:__exception__, :__struct__])
+      |> Map.keys()
+
+    only = Keyword.get(options, :only, fields)
+    except = Keyword.get(options, :except, [])
+
+    filtered_fields =
+      fields
+      |> Enum.reject(&(&1 in except))
+      |> Enum.filter(&(&1 in only))
+
+    inspect_fun =
+      case {only, except} do
+        {^fields, []} -> quote(do: Inspect.Map.inspect(map, name, opts))
+        _ -> quote(do: Inspect.Any.inspect(map, name, opts))
+      end
```

Comment:
Maybe rewrite as:

```elixir
inspect_module = 
  if fields == only and except == [] do
    quote(do: Inspect.Map)
  else
    quote(do: Inspect.Any)
  end

# and then
unquote(inspect_module).inspect(map, name, opts)
```

which also makes it cleaner by not using variables in the quote before they are defined. Thoughts?

Reply:
Agree. Done in https://github.com/elixir-lang/elixir/pull/8104/commits/c4c59c8f3f8ed1b15168fac45829f953b0c0a91e